routes/drinks.js

```diff
@@ -0,0 +1,38 @@
+const { check, validationResult } = require("express-validator");
+const router = require("express").Router();
+const db = require("../models");
+
+const allowList = ["coffee", "tea", "beer", "wine"];
+
+router.get(
+  "/id/:id",
+  [check("id", "malformed ID").isLength({ max: 24, min: 24 })],
+  async (req, res) => {
+    const errors = validationResult(req);
+    if (!errors.isEmpty()) {
+      return res.status(400).json({
+        errors: errors.array(),
+      });
+    }
+
+    const drink = await db.Drinks.findById(req.params.id);
+    res.json(drink);
+  }
+);
+
+router.get("/:type", async (req, res) => {
+  const { type } = req.params;
+  if (!allowList.includes(type)) {
+    return res.json([]);
+  }
+
+  const drinks = await db.Drinks.find({ type });
```

Comment:
Not sure if this will return a cursor or an array. I know with the driver you have to .toArray() it, but maybe that's not the case with mongoose